### PR TITLE
fix(changelog): Escape # in milestones

### PR DESCRIPTION
**Pull request by BYK**
We need to escape the `#` sign in milestone titles and bodies as it may interfere or get hidden with the markdown formatting (it uses `#` to identify headers). This patch fixes the issue and it also fixes another issue when the milestone description was empty.

### CODE REVIEW

File: src/utils/__tests__/changelog.test.ts [hunk 1]
```diff
@@ -487,8 +487,7 @@ describe('generateChangesetFromGit', () => {
         },
         M5: {
           title: 'Better driver experience',
-          description:
-            'We are working on making your driving experience more pleasant and safer.',
+          description: '',
```
Comment (chadwhitacre) [inline src/utils/__tests__/changelog.test.ts:490]: Even more of an edge case (and not necessary for this PR) ... what if the description is `'  '` ... i.e., non-empty whitespace?

File: src/utils/__tests__/changelog.test.ts [hunk 2]
```diff
@@ -513,4 +510,69 @@ describe('generateChangesetFromGit', () => {
...
+      {
+        hash: 'abcdef1234567890',
+        pr: '123',
+        milestone: '1',
+      },
+    ]);
+    mockClient.mockResolvedValueOnce({
+      repository: {
+        M1: {
+          title: 'Drivetrain #1 in town',
```
Comment (chadwhitacre) [inline src/utils/__tests__/changelog.test.ts:532]: 😂

File: src/utils/changelog.ts [hunk 3]
```diff
@@ -39,9 +39,13 @@ export interface ChangesetLoc {
...
 }
 
+function escapeMarkdownPound(text: string): string {
+  return text.replace(/#/g, '&#35;');
+}
+
 function markdownHeader(level: number, text: string): string {
   const prefix = new Array(level + 1).join('#');
-  return `${prefix} ${text}`;
+  return `${prefix} ${escapeMarkdownPound(text)}`;
```
Comment (chadwhitacre) [inline src/utils/changelog.ts:48]: YOU MEAN OCTOTHORPE :trollface:

**Review by chadwhitacre — APPROVED** [review]

File: src/utils/__tests__/changelog.test.ts (hunk 1 above)
Comment (BYK) [reply src/utils/__tests__/changelog.test.ts:490]: We can trim it but I'd assume GitHub would do that already.

File: src/utils/changelog.ts (hunk 3 above)
Comment (BYK) [reply src/utils/changelog.ts:48]: LOL WUT?

Comment (chadwhitacre) [reply src/utils/changelog.ts:48]: https://en.wikipedia.org/wiki/Number_sign 😁

Screen Shot 2021-09-24 at 10 07 31 AM
